en: update doc for deploying TiDB cluster on AWS EKS (#49)
pingcap/docs-tidb-operator: en/deploy-on-aws-eks.md

>
> EKS versions earlier than 1.14 do not support auto enabling cross-zone load balancing via Network Load Balancer (NLB). Therefore, unbalanced pressure distributed among TiDB instances can be expected in default settings. It is strongly recommended that you refer to [AWS Documentation](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-disable-crosszone-lb.html#enable-cross-zone) to manually enable cross-zone load balancing for a production environment.

### Deploy TiDB cluster and monitor

1. Prepare the TidbCluster and TidbMonitor CR files:

    {{< copyable "shell-regular" >}}

    ```shell
    cd manifests/ && mv db-monitor.yaml.example db-monitor.yaml && mv db.yaml.example db.yaml
    ```

    To complete the CR file configuration, refer to [API documentation](https://github.com/pingcap/tidb-operator/blob/master/docs/api-references/docs.html).

    > **Note:**
    >

>     > * Replace all `CLUSTER_NAME` in `dm.yaml` and `db-monitor.yaml` files with `default_cluster_name` configured during EKS deployment.

Reviewer: ```suggestion
    > * Replace all `CLUSTER_NAME` in `db.yaml` and `db-monitor.yaml` files with `default_cluster_name` configured during EKS deployment.
```

## Upgrade

To upgrade the TiDB cluster, edit the `spec.version` by `kubectl --kubeconfig credentials/kubeconfig_<eks_name> edit tc <default_cluster_name> -n <namespace>`.

> The upgrading doesn't finish immediately. You can watch the upgrading process by `kubectl --kubeconfig credentials/kubeconfig_<eks_name> get po -n <namespace> --watch`.

Reviewer: ```suggestion
The upgrading doesn't finish immediately. You can watch the upgrading progress by `kubectl --kubeconfig credentials/kubeconfig_<eks_name> get po -n <namespace> --watch`.
```

## Scale

To scale the TiDB cluster, modify the `default_cluster_tikv_count` or `default_cluster_tidb_count` variable in the `terraform.tfvars` file to your desired count, and then run `terraform apply` to scale out the number of the corresponding component nodes.

After the scaling, modify the `replicas` of the corresponding component by the following command:

> 

Reviewer: ```suggestion

{{< copyable "shell-regular" >}}

```

```

> `kubectl --kubeconfig credentials/kubeconfig_<eks_name> edit tc <default_cluster_name> -n <namespace>`

Reviewer: ```suggestion
kubectl --kubeconfig credentials/kubeconfig_<eks_name> edit tc <default_cluster_name> -n <namespace>
```

```

For example, to scale out the TiDB nodes, you can modify the number of TiDB instances from 2 to 4:

```hcl
default_cluster_tidb_count = 4
```

> After the scaling, modify the `spec.tidb.replicas` in `TidbCluster` to scale out the Pod.

Reviewer 2: ```suggestion
After the nodes scaling out, modify the `spec.tidb.replicas` in `TidbCluster` to scale out the Pod.
```

> **Note:**
>

> > Currently, scaling in is NOT supported because we cannot determine which node to scale.

Reviewer 2: ```suggestion
> Currently, scaling in is NOT supported because we cannot determine which node will be scaled in.
```

> > Scaling out needs a few minutes to complete, you can watch the scaling out by `kubectl --kubeconfig credentials/kubeconfig_<eks_name> get po -n <namespace> --watch`.

Reviewer: ```suggestion
> Scaling out needs a few minutes to complete, you can watch the scaling out progress by `kubectl --kubeconfig credentials/kubeconfig_<eks_name> get po -n <namespace> --watch`.
```

### Customize TiDB Operator

To customize the TiDB Operator, modify the `operator_values` parameter in `terraform.tfvars` to pass the customized contents of `values.yaml`. For example:

```hcl
operator_values = "./operator_values.yaml"
```

## Manage multiple TiDB clusters

An instance of `tidb-cluster` module corresponds to a TiDB cluster in the EKS cluster. If you want to create a node pool for a new TiDB cluster, edit `./cluster.tf` and add a new instance of `tidb-cluster` module:

```hcl
module example-cluster {
  source = "../modules/aws/tidb-cluster"
  eks = local.eks
  subnets = local.subnets
  region = var.region
  cluster_name    = "example"
  ssh_key_name                  = module.key-pair.key_name
  pd_count                      = 1
  pd_instance_type              = "c5.large"
  tikv_count                    = 1
  tikv_instance_type            = "c5d.large"
  tidb_count                    = 1
  tidb_instance_type            = "c4.large"
  monitor_instance_type         = "c5.large"
  create_tidb_cluster_release   = false
}
```

> **Note:**
>
> The `cluster_name` of each cluster must be unique.

> When you finish modification, execute `terraform init` and `terraform apply` to create the TiDB cluster.

Reviewer 2: ### 
```suggestion
When you finish the modification, execute `terraform init` and `terraform apply` to create the node pools for the TiDB cluster.
```

The Terraform module in our case typically combines several sub-modules:

> - A `tidb-operator` module, that creates the EKS cluster and [deploy TiDB Operator](deploy-tidb-operator.md) on the EKS cluster

Reviewer: ```suggestion
- A `tidb-operator` module, which creates the EKS cluster and [deploy TiDB Operator](deploy-tidb-operator.md) on the EKS cluster
```

> - A `tidb-cluster` module, that creates the resource pool required by the TiDB cluster

Reviewer: ```suggestion
- A `tidb-cluster` module, which creates the resource pool required by the TiDB cluster
```
